### PR TITLE
[SPARK-25282][K8s][DOC] Improved docs to avoid running into InvalidClassException.

### PR DESCRIPTION
**Pull request by ScrapCodes**
## What changes were proposed in this pull request?

Documentation changes, on client mode. A user needs to make a distribution before he can use client mode.

## How was this patch tested?
Doc improvement comments are based on running locally. The patch itself is a document patch, so may not need testing.

### CODE REVIEW

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -123,6 +123,20 @@ Starting with Spark 2.4.0, it is possible to run Spark applications on Kubernete
 runs in client mode, the driver can run inside a pod or on a physical host. When running an application in client mode,
 it is recommended to account for the following factors:
 
+### Make a distribution
+
+In the case you are working with latest code directly from the git code base of Apache Spark, by building from the source.
```
Comment (liyinan926) [inline docs/running-on-kubernetes.md]: Nit: `the latest code` and `a docker repository` below. Also please use `Spark` with a capital `S` consistently.

**Review by liyinan926 — COMMENTED** [review]
Some minor comment, otherwise LGTM.

**Comment by liyinan926** [comment]
BTW: just realized that the image build script at `bin/docker-image-tool.sh` works both in a source directory and in an unzipped distribution. So you shouldn't need to make a distribution to build an image.

File: docs/running-on-kubernetes.md (hunk 1 above)
Comment (srowen) [reply docs/running-on-kubernetes.md]: This and the sentence below aren't quite grammatical. This also kind of duplicates the docs on how to build a release. Why not just say something like:

When testing against a custom build of Spark, from the latest source code or custom modifications, it's necessary to create a release and deploy images to a Docker repository. See the [Docker images] section, and [docs on releasing]. Example:

... example command
```

**Comment by srowen** [comment]
What does this have to do with InvalidClassException by the way?

**Pull request closed without merging**